### PR TITLE
Alecb/2022 10 14 preview powershell release

### PR DESCRIPTION
**Pull request by alec-baird**
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by alec-baird** [comment]
These tests seem to be failing due to a framework issue, as the tests themselves are passing locally and the missing request talked about seems to just be one number off, instead of a +2 in the recording it is a +3.

File: src/DesktopVirtualization/Az.DesktopVirtualization.psd1 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 @{
   GUID = 'c78eb738-b339-4296-8c9f-13ef28817c3c'
   RootModule = './Az.DesktopVirtualization.psm1'
-  ModuleVersion = '2.1.0'
+  ModuleVersion = '2.2.0'
```
Comment (vidai-msft) [inline src/DesktopVirtualization/Az.DesktopVirtualization.psd1:4]: Please do NOT update the version. We will bump during release process.

File: src/DesktopVirtualization/README.md [hunk 2]
```diff
@@ -47,15 +47,15 @@ In this directory, run AutoRest:
...
 require:
   - $(this-folder)/../readme.azure.noprofile.md
 sanitize-names: true
 subject-prefix: 'Wvd'
 input-file:
-- $(repo)/specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/stable/2022-09-09/desktopvirtualization.json
+- $(repo)/specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2022-10-14-preview/desktopvirtualization.json
 
-module-version: 2.1.0
+module-version: 2.2.0
```
Comment (vidai-msft) [inline src/DesktopVirtualization/README.md:58]: Please do NOT update the version. We will bump during release process.

**Review by vidai-msft — CHANGES_REQUESTED** [review]

**Comment by vidai-msft** [comment]
> These tests seem to be failing due to a framework issue, as the tests themselves are passing locally and the missing request talked about seems to just be one number off, instead of a +2 in the recording it is a +3.

@alec-baird this is not an issue but a limitation. The test framework for Autorest cmdlets does not support recording cross-module cmdlets. Can you please move the cmdlets invocation that do not belong to this module to the function setupEnv in utils.ps1?

**Comment by vidai-msft** [comment]
@alec-baird please leave a comment here as change log for this PR.

**Comment by alec-baird** [comment]
> @alec-baird please leave a comment here as change log for this PR.

Changelog:
Added Get-AzWvdPrivateLinkResource, Get-AzWvdPrivateLinkConnection, and Remove-AzWvdPrivateLinkConnection cmdlets.

**Comment by alec-baird** [comment]
> > These tests seem to be failing due to a framework issue, as the tests themselves are passing locally and the missing request talked about seems to just be one number off, instead of a +2 in the recording it is a +3.
> 
> @alec-baird this is not an issue but a limitation. The test framework for Autorest cmdlets does not support recording cross-module cmdlets. Can you please move the cmdlets invocation that do not belong to this module to the function setupEnv in utils.ps1?

The issue with that implementation is that we only want the cross-module cmdlet to run on these few tests, as it is a lengthy cmdlet runtime (almost a minute per call). Can we get a test exception here instead?

**Comment by vidai-msft** [comment]
@alec-baird please remember to fix the test issue in CI. Thanks!

**Pull request merged**